include/bluetooth/mesh/light_ctl_temp_srv.h

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file
+ * @defgroup bt_mesh_light_ctl_temp_srv Light CTL model
+ * @{
+ * @brief API for the Light CTL Temprature Server model.
+ */
+
+#ifndef BT_MESH_LIGHT_CTL_TEMP_SRV_H__
+#define BT_MESH_LIGHT_CTL_TEMP_SRV_H__
+
+#include <bluetooth/mesh/light_ctl.h>
+#include <bluetooth/mesh/model_types.h>
+#include <bluetooth/mesh/gen_lvl_srv.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct bt_mesh_light_ctl_temp_srv;
+
+/** @def BT_MESH_LIGHT_CTL_TEMP_SRV_INIT
+ *
+ * @brief Initialization parameters for a @ref bt_mesh_light_ctl_temp_srv
+ * instance.
+ *
+ * @param[in] _handlers Light CTL Temprature Server callbacks.
+ */
+#define BT_MESH_LIGHT_CTL_TEMP_SRV_INIT(_handlers)                             \
+	{                                                                      \
+		.handlers = _handlers,                                         \
+		.lvl = BT_MESH_LVL_SRV_INIT(                                   \
+			&_bt_mesh_ctl_temp_srv_lvl_handlers),                  \
+		.pub = { .msg = NET_BUF_SIMPLE(BT_MESH_MODEL_BUF_LEN(          \
+				 BT_MESH_LIGHT_CTL_TEMP_STATUS,                \
+				 BT_MESH_LIGHT_CTL_MSG_MAXLEN_TEMP_STATUS)) }, \
+		.temp_range = {                                                \
+			.min = BT_MESH_LIGHT_CTL_TEMP_RANGE_MIN,               \
+			.max = BT_MESH_LIGHT_CTL_TEMP_RANGE_MAX                \
+		}                                                              \
+	}
+
+/** @def BT_MESH_MODEL_LIGHT_CTL_TEMP_SRV
+ *
+ * @brief Light CTL Temprature Server model composition data entry.
+ *
+ * @param[in] _srv Pointer to a @ref bt_mesh_light_ctl_temp_srv instance.
+ */
+#define BT_MESH_MODEL_LIGHT_CTL_TEMP_SRV(_srv)                                 \
+	BT_MESH_MODEL_LVL_SRV(&(_srv)->lvl),                                   \
+		BT_MESH_MODEL_CB(BT_MESH_MODEL_ID_LIGHT_CTL_TEMP_SRV,          \
+				 _bt_mesh_light_ctl_temp_srv_op, &(_srv)->pub, \
+				 BT_MESH_MODEL_USER_DATA(                      \
+					 struct bt_mesh_light_ctl_temp_srv,    \
+					 _srv),                                \
+				 &_bt_mesh_light_ctl_temp_srv_cb)
+
+/** Set structure for CTL Temprature messages */
+struct bt_mesh_light_ctl_temp_cb_set {
+	uint16_t *temp;
+	int16_t *delta_uv;
+	const struct bt_mesh_model_transition *transition;
+};
+
+/** Light CTL Temprature Server state access handlers. */
+struct bt_mesh_light_ctl_temp_srv_handlers {
+	/** @brief Set the CTL Temprature state.
+	 *
+	 * @note This handler is mandatory.
+	 *
+	 * @param[in] srv Server to set the CTL Temprature state of.
+	 * @param[in] ctx Message context.
+	 * @param[in] set Parameters of the state change.
+	 * @param[out] rsp Response structure to be filled.
```

Comment:
Document that this can be NULL

Reply:
Can you elaborate? Not quite sure what you mean

Reply:
Sorry, I was a bit too fast on this one